internal/agent/device/lifecycle/manager.go

```diff
@@ -92,12 +99,78 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		if err := m.updateWithStartedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission started status: %w", err)
+		}
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// TODO: add support for additional decommissioning target types.
+		// these are the steps that will take places between Started and Completed status
+
+		if err := m.updateWithCompletedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission completed status: %w", err)
+		}
+		m.log.Warn("No errors during decommissioning, updated Condition to decommissioning completed")
+
+		// after this point the device will no longer be able to communicate with the management service
+		m.log.Warn("Preparing to wipe agent certificate and keys and reboot")
+		return m.wipeAndReboot(ctx)
+	}
+	return nil
+}
+
+func (m *LifecycleManager) updateWithStartedCondition(ctx context.Context) error {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateStarted),
+		Message: "The device has started decommissioning",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+		return fmt.Errorf("failed to update decommission started status: %w", updateErr)
+	}
+	return nil
+}
+
+func (m *LifecycleManager) updateWithCompletedCondition(ctx context.Context) error {
+	updateErr := m.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
+		Type:    v1alpha1.DeviceDecommissioning,
+		Status:  v1alpha1.ConditionStatusTrue,
+		Reason:  string(v1alpha1.DecommissionStateComplete),
+		Message: "The device has completed decommissioning and will wipe its management certificate",
+	})
+	if updateErr != nil {
+		m.log.Warnf("Failed setting status: %v", updateErr)
+		return fmt.Errorf("failed to update decommission completed status: %w", updateErr)
+	}
+	return nil
+}
+
+// point of no return - wipes management cert and keys
+func (m *LifecycleManager) wipeAndReboot(ctx context.Context) error {
+	err := m.deviceReadWriter.RemoveFile(m.managementCertPath)
```

Comment:
Why delete only the management cert and not the management key (which is what's required impersonate a device with the same fingerprint)?

Reply:
Good catch, I missed this since it was not in the lifecycle manager but I've updated now to pass this in and remove the file.

Reply:
if one fails, logically do you want to continue and remove as much as you can or return?

Reply:
We should remove as much as possible. There isn't a good way to do troubleshooting at this stage

Reply:
We should add an wipe method to deviceReaderWriter, that must store 0's over all the contents, then remove file.

This is not perfect and best efforts, in some filesystems/or when using lvm with snapshots, the block changes will be logged, and the older content is still persisted to disk.

What I am trying to prevent here is that somebody would scan the disk raw contents for keys/certs, and find ours this is actually something relatively easy to do technically.

Reply:
Another question, will we incorporate support for the user running additional wipe scripts? 

i.e. customers may want to perform full disk block wipe under some conditions.

Reply:
That's a great idea @mangelajo , I'll make an update.
Re: your question, yes, supporting additional scripts and options is future functionality that we will add when we add additional decommissioning "targets".